src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2554,14 +2554,34 @@ float GameObject::GetInteractionDistance() const
 {
     switch (GetGoType())
     {
-        /// @todo find out how the client calculates the maximal usage distance to spellless working
-        // gameobjects like guildbanks and mailboxes - 10.0 is a just an abitrary choosen number
-        case GAMEOBJECT_TYPE_GUILD_BANK:
-        case GAMEOBJECT_TYPE_MAILBOX:
+        case GAMEOBJECT_TYPE_AREADAMAGE:
+            return 0.0f;
+        case GAMEOBJECT_TYPE_QUESTGIVER:
+        case GAMEOBJECT_TYPE_TEXT:
+        case GAMEOBJECT_TYPE_FLAGSTAND:
+        case GAMEOBJECT_TYPE_FLAGDROP:
+        case GAMEOBJECT_TYPE_MINI_GAME:
+            return 5.5555553f;
+        case GAMEOBJECT_TYPE_BINDER:
             return 10.0f;
+        case GAMEOBJECT_TYPE_CHAIR:
+        case GAMEOBJECT_TYPE_BARBER_CHAIR:
+            return 3.0f;
         case GAMEOBJECT_TYPE_FISHINGHOLE:
         case GAMEOBJECT_TYPE_FISHINGNODE:
             return 20.0f + CONTACT_DISTANCE; // max spell range
```

Comment:
I took all the values from the client executable, so you might as well have left 100 yards there.
![image](https://user-images.githubusercontent.com/43143822/56990251-4f8a1e00-6b9d-11e9-889e-2cd9ef8bdcc6.png)
![image](https://user-images.githubusercontent.com/43143822/56990264-56189580-6b9d-11e9-91ea-c771895169c6.png)
`GAMEOBJECT_TYPE_FISHINGNODE` is initialized to 100 yards, but `GAMEOBJECT_TYPE_FISHINGHOLE` reuses the same constructor as `GAMEOBJECT_TYPE_SPELL_FOCUS` (which is also reused by `GAMEOBJECT_TYPE_DUEL_ARBITER` and `GAMEOBJECT_TYPE_AURA_GENERATOR`, hey, compiler can optimize things however it wants) which just calls the base constructor and initializes interaction distance to default 5 yards. Which is because player never really interacts with a fishing hole, only with a fishing node, which on the server checks if there are any holes nearby and sends their loot instead of the fishing loot for the current zone.

But I won't insist, since realistically the only time this will be relevant is if a player decides to fish from a moving transport, like a ship or a turtle taxi in Northrend. 😄